### PR TITLE
Fix the File importer and Database Importer `Import` action.

### PR DESCRIPTION
**Pull request by Auriga2**
### Description of the Change
The File importer now obeys the "files include headers" option when importing and when running the plugin through API. Database Importer now imports all rows when the `Import` button is clicked.

### Why Should This Be In Core?
Bug fix

### Benefits
Bug fix

### Verification Process
1) Create a CSV file with no header row e.g.:
5,6,7,2009-12-05 23:59:59,2009-12-1
1,1,1,2009-12-15 23:59:59,2009-12-12
a,a,a,2009-12-25 23:59:59,2009-12-31

2) Click File/Import/From File

3) Disable the Files include headers check box and Select the file to import using green + sign

4) Verify the column header are inserted in the preview pane

5) Drag and drop Source Node-Identifier attribute on to the Col1 (And more attributes if you like)

6) import it by clicking `Import` button at the bottom

7) Verify all 4 nodes are drawn on a graph and verify the confirmation pop up message displays that the 4 rows are imported instead of 3

8) Verify it doesn't  import/draw the first row when the Option is checked

9) Perform an import using the Database Importer, and verify it's importing/drawing all the rows including the first row when the `Import` button is clicked.

### Applicable Issues
https://github.com/constellation-app/constellation/issues/1317

### CODE REVIEW

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java [hunk 1]
```diff
@@ -112,6 +113,12 @@ public PluginParameters createParameters() {
         parserParameters.setDescription("The PluginParameters used by the parser");
         params.addParameter(parserParameters);
 
+        final PluginParameter<BooleanParameterValue> filesIncludeHeadersParam = BooleanParameterType.build(FILES_INCLUDE_HEADERS_PARAMETER_ID);
+        filesIncludeHeadersParam.setName("files Include Headers");
```
Comment (arcturus2) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:117]: I'd prefer we start with a capitalised word (i.e "Files") but its interesting to see the GUI has it capitalised.

**Review by arcturus2 — APPROVED** [review]

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ConfigurationPane.java [hunk 2]
```diff
@@ -191,12 +191,12 @@ public void setSampleData(final String[] columnLabels, final List<String[]> curr
      * @return A List&lt;ImportDefinition&gt; where each list element
      * corresponds to a RunPane tab.
      */
-    public List<ImportDefinition> createDefinitions() {
+    public List<ImportDefinition> createDefinitions(boolean isFilesIncludeHeadersEnabled) {
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ConfigurationPane.java:194]: parameter final

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ImportController.java [hunk 3]
```diff
@@ -372,8 +372,8 @@ public String showSetDefaultValueDialog(final String attributeName, final String
      * @return A List&lt;ImportDefinition&gt; where each list element
      * corresponds to a RunPane tab.
      */
-    public List<ImportDefinition> getDefinitions() {
-        return configurationPane.createDefinitions();
+    public List<ImportDefinition> getDefinitions(boolean isFilesIncludeHeadersEnabled) {
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ImportController.java:375]: similar

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java (hunk 1 above)
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:117]: Use captial F for files

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java [hunk 4]
```diff
@@ -112,6 +113,12 @@ public PluginParameters createParameters() {
...
+        filesIncludeHeadersParam.setDescription("True if the files include headers in the first row, default is True");
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:118]: I'd probably omit everything after the comma as the default value doesn't really need an explicit mention here

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java [hunk 5]
```diff
@@ -176,21 +183,24 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
         final List<ImportDefinition> definitions = (List<ImportDefinition>) parameters.getParameters().get(DEFINITIONS_PARAMETER_ID).getObjectValue();
         final Boolean initialiseWithSchema = parameters.getParameters().get(SCHEMA_PARAMETER_ID).getBooleanValue();
         final PluginParameters parserParameters = (PluginParameters) parameters.getParameters().get(PARSER_PARAMETER_IDS_PARAMETER_ID).getObjectValue();
+        final Boolean filesIncludeHeaders = parameters.getParameters().get(FILES_INCLUDE_HEADERS_PARAMETER_ID).getBooleanValue();
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:186]: use primitive `boolean` here

**Review by antares1470 — CHANGES_REQUESTED** [review]
Tested it as well and it seems to be fixed.

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java (hunk 1 above)
Comment (Auriga2) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:117]: Oh my bad, but I guess this is not really used in the GUI (that's why it worked before the changes here) and should be used in the API if we ever expose the parameters.

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java (hunk 4 above)
Comment (Auriga2) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:118]: This option is set to True by default, does this have a different meaning?

**Comment by Auriga2** [comment]
I guess this is not used in the GUI anyway, but it'll be, if we expose them
through the API someday.

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java (hunk 4 above)
Comment (Auriga2) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:118]: I guess this is not used in the GUI anyway, but it'll be, if we expose them through the API someday.

Comment (antares1470) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:118]: Something like that yes

Comment (Auriga2) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java:118]: Fixed for the schemaParam as well, to be consistent...

**Review by arcturus2 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**